aioftp/client.py

```diff
@@ -368,16 +370,25 @@ def parse_ls_date(self, s, *, now=None):
             try:
                 if now is None:
                     now = datetime.datetime.now()
-                d = datetime.datetime.strptime(s, "%b %d %H:%M")
-                d = d.replace(year=now.year)
-                diff = (now - d).total_seconds()
-                if diff > HALF_OF_YEAR_IN_SECONDS:
-                    d = d.replace(year=now.year + 1)
-                elif diff < -HALF_OF_YEAR_IN_SECONDS:
-                    d = d.replace(year=now.year - 1)
+                if s.startswith('Feb 29'):
+                    # Need to find the nearest leap year
+                    year = now.year
+                    while not calendar.isleap(year):
+                        year -= 1
```

Comment:
Leap year can be from future half of the year. So this logic will find wrong (past) leap year.

Reply:
Yes, look at the changes, please.